### PR TITLE
Improved help without table library

### PR DESCRIPTION
**Pull request by potatowagon**
to pass travis. With the neat table library would be PR #163

### CODE REVIEW

File: src/test/java/seedu/address/ui/HelpWindowTest.java [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
 
-import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.*;
 
 import org.junit.After;
 import org.junit.Before;
+
+//import org.junit.After;
+//import org.junit.Before;
+
```
Comment (pyokagan) [inline src/test/java/seedu/address/ui/HelpWindowTest.java]: ????????

**Comment by pyokagan** [comment]
@potatowagon How is this ready? Travis is failing.

**Comment by pyokagan** [comment]
Ugh, this is not mergeable for now. Your code structure needs to be reworked to not call `showAndWait()` in the constructor, or else the help window cannot be tested.

**Comment by pyokagan** [comment]
Also, go look up the difference between monospace fonts and non-monospace fonts

**Comment by pyokagan** [comment]
Maybe it would be easier for everyone if you just use the web browser instead of trying to recreate tables in alert dialogs? This way you will have the full power of HTML to layout the content. Maybe build off this?

``` diff
diff --git a/src/main/java/seedu/address/ui/HelpWindow.java b/src/main/java/seedu/address/ui/HelpWindow.java
index 2831e49..3881df1 100644
--- a/src/main/java/seedu/address/ui/HelpWindow.java
+++ b/src/main/java/seedu/address/ui/HelpWindow.java
@@ -4,6 +4,7 @@ import javafx.fxml.FXML;
 import javafx.scene.web.WebView;
 import javafx.stage.Modality;
 import javafx.stage.Stage;
+import seedu.address.MainApp;
 import seedu.address.commons.util.FxViewUtil;

 /**
@@ -13,8 +14,7 @@ public class HelpWindow extends UiPart<Stage> {

     private static final String ICON = "/images/help_icon.png";
     private static final String FXML = "/view/HelpWindow.fxml";
-    private static final String USERGUIDE_URL =
-            "https://github.com/se-edu/addressbook-level4/blob/master/docs/UserGuide.md";
+    private static final String USERGUIDE_HTML = "/view/HelpWindow.html";

     @FXML
     private WebView webView;
@@ -23,7 +23,7 @@ public class HelpWindow extends UiPart<Stage> {
         super(FXML);
         getRoot().initModality(Modality.WINDOW_MODAL);
         FxViewUtil.setStageIcon(getRoot(), ICON);
-        webView.getEngine().load(USERGUIDE_URL);
+        webView.getEngine().load(MainApp.class.getResource(USERGUIDE_HTML).toExternalForm());
     }

 }
diff --git a/src/main/resources/view/HelpWindow.html b/src/main/resources/view/HelpWindow.html
index e69de29..71c509c 100644
--- a/src/main/resources/view/HelpWindow.html
+++ b/src/main/resources/view/HelpWindow.html
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="utf-8" />
+    <title>Help</title>
+</head>
+<body>
+    <h1>WOW Tables!</h1>
+    <table style="width:100%" border="1">
+    <tr>
+        <td><code>WOW</code></td>
+        <td><code>WOW</code></td>
+    </tr>
+    </table>
+</body>
+</html>
diff --git a/src/test/java/seedu/address/ui/HelpWindowTest.java b/src/test/java/seedu/address/ui/HelpWindowTest.java
index 13a8158..999e895 100644
--- a/src/test/java/seedu/address/ui/HelpWindowTest.java
+++ b/src/test/java/seedu/address/ui/HelpWindowTest.java
@@ -2,6 +2,8 @@ package seedu.address.ui;

 import static org.junit.Assert.assertEquals;

+import java.net.URL;
+
 import org.junit.After;
 import org.junit.Before;
 import org.junit.BeforeClass;
@@ -11,6 +13,7 @@ import org.testfx.api.FxRobot;
 import org.testfx.api.FxToolkit;

 import javafx.scene.web.WebView;
+import seedu.address.MainApp;
 import seedu.address.testutil.GuiTests;

 @Category({GuiTests.class})
@@ -41,9 +44,9 @@ public class HelpWindowTest extends FxRobot {
     }

     @Test
-    public void constructor() {
-        assertEquals("https://github.com/se-edu/addressbook-level4/blob/master/docs/UserGuide.md",
-                     webView.getEngine().getLocation());
+    public void constructor() throws Exception {
+        assertEquals(MainApp.class.getResource("/view/HelpWindow.html"),
+                     new URL(webView.getEngine().getLocation()));
     }

 }
```

**Comment by pyokagan** [comment]
@potatowagon So what's the status of this? Is this ready?

**Comment by potatowagon** [comment]
ive not done JUnit test, but testing with the UI works

**Comment by pyokagan** [comment]
Okay, try adding tests. Our test coverage really needs to improve.

**Comment by potatowagon** [comment]
ok

**Comment by pyokagan** [comment]
@potatowagon Very important: Courier New is a commercial font and we definitely CANNOT bundle it with our application. Maybe try something like [Roboto Mono](https://fonts.google.com/specimen/Roboto+Mono), and remember to bundle the LICENSE file as well.

**Comment by potatowagon** [comment]
ok,

**Comment by potatowagon** [comment]
GUI test written, Courier New Bundle removed.

**Comment by potatowagon** [comment]
ready to merge

File: src/main/java/seedu/address/ui/HelpWindow.java [hunk 2]
```diff
@@ -1,29 +1,100 @@
...
+
+        //window settings
+        stage.setAlwaysOnTop(true);
+        helpBox.initModality(Modality.NONE);
+        helpBox.setResizable(true);
+        helpBox.getDialogPane().setPrefWidth(800.0);
+        helpBox.getDialogPane().setPrefHeight(330.0);
+        //helpBox.getDialogPane().getStylesheets().add("src/main/resources/view/DarkTheme.css");
+        //helpBox.getDialogPane().getStyleClass().add("helpBox");
+        helpBox.getDialogPane().setStyle("-fx-font-family: \"Courier New\";" + "-fx-font-weight: 700;" + "font-style: normal;" + "-fx-background-color: white;" + "-fx-color: black;");
```
Comment (pyokagan) [inline src/main/java/seedu/address/ui/HelpWindow.java:67]: Hmm, this still reads "Courier New"...

Comment (pyokagan) [reply src/main/java/seedu/address/ui/HelpWindow.java:67]: When I replace it with the equivalent of using `DarkTheme.css` though, I can't get the Roboto Mono font to load :-/ Hmmm, I think JavaFX's support for `@font-face` just sucks.

**Review by pyokagan — APPROVED** [review]
Okay, I fixed the fonts problem on top. Will merge.

**Comment by pyokagan** [comment]
Merged as  ef673e113227e3a31ff16a17f94cc34cec00ee0b

**Comment by potatowagon** [comment]
thankyou :)

**Pull request closed without merging**